tests/cypress/tests-e2e/legacyParamImport.cy.js

```diff
@@ -411,4 +440,142 @@ describe('Test on legacy param import', () => {
             cy.readStoreValue('state.position.projection.epsgNumber').should('eq', 3857)
         })
     })
+
+    context('Extra Parameter Imports', () => {
+        it('sets the language correctly', () => {
+            cy.goToMapView({ lang: 'it' }, false)
+            cy.readStoreValue('state.i18n.lang').should('eq', 'it')
+        })
+
+        it('shows the correct crosshair', () => {
+            function checkCrosshair(crossHair) {
+                cy.readStoreValue('state.position').then((positionStore) => {
+                    expect(positionStore.crossHairPosition).to.eql(positionStore.center)
+                    switch (crossHair) {
+                        case 'cross':
+                            expect(positionStore.crossHair).to.eq(CrossHairs.cross)
+
+                            break
+                        case 'circle':
+                            expect(positionStore.crossHair).to.eq(CrossHairs.circle)
+
+                            break
+                        case 'bowl':
+                            expect(positionStore.crossHair).to.eq(CrossHairs.bowl)
+
+                            break
+                        case 'point':
+                            expect(positionStore.crossHair).to.eq(CrossHairs.point)
+
+                            break
+                        case 'marker':
+                            expect(positionStore.crossHair).to.eq(CrossHairs.marker)
+
+                            break
+                    }
+                })
+            }
+            const legacyCrossHairTypes = ['cross', 'circle', 'bowl', 'point', 'marker']
+            legacyCrossHairTypes.forEach((crossHair) => {
+                cy.goToMapView({ crosshair: crossHair }, false)
+                checkCrosshair(crossHair)
+            })
+        })
+
+        it('shows the compare slider at the correct position', () => {
+            cy.goToMapView(
+                {
+                    layers: 'test-1.wms.layer',
+                    swipe_ratio: '0.3',
+                },
+                false
+            )
+            // initial slider position is width * 0.3 -20
+            cy.get('[data-cy="compare_slider"]').then((slider) => {
+                cy.readStoreValue('state.ui.width').then((width) => {
+                    cy.wrap(slider.position()['left']).should('eq', width * 0.3 - 20)
+                })
```

Comment:
Use `.should((width)=>{})` construct instead with `expect` this ensure retries until the assertion is true